README.md

```diff
@@ -319,77 +322,204 @@ spec:
+    plan: sample-plan // The service plan name
+    type: sample-service // The service offering name
 ```
 
 ###### Credentials and Service Info as One JSON Object
-To show both credentials returned from the broker and service instance info as a JSON object, use the 'secretRootKey' attribute in the service binding spec.
 
-The value of 'secretRootKey' is the name of the key that stores both credentials and serivce instance info in JSON format.
+To show both credentials returned from the broker and additional `ServiceInstance` attributes as a JSON object, use the 'secretRootKey' attribute in the `ServiceBinding` spec.
+
+The value of 'secretRootKey' is the name of the key that stores both credentials and `ServiceInstance` info in JSON format.
 
-```bash
-your-secretRootKey-value:
+Input:
+
+```yaml
+apiVersion: services.cloud.sap.com/v1
+kind: ServiceBinding
+metadata:
+  name: sample-binding
+spec:
+  serviceInstanceName: sample-instance
+  secretRootKey: your-secretRootKey-value
+```
+Output:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: sample-binding
+stringData:
+    your-secretRootKey-value:
+    {
+        uri: https://my-service.authentication.eu10.hana.ondemand.com,
+        client_id: admin,
+        client_secret: ********,
+        instance_guid: your-sample-instance-guid, // The service instance id
+        instance_name: sample-instance-name, // Taken from the service instance external_name field if set. Otherwise from metadata.name
+        plan: sample-instance-plan, // The service plan name
+        type: sample-instance-offering, // The service offering name
+    }
+```
+###### Custom Templates
+
+For additional flexibility, you can also model your `Secret` resources according to your specific needs.
+To generate a custom-formatted `Secret`, use the `secretTemplate` attribute in the `ServiceBinding` spec.
+The value of `secretTemplate` must be a Go template. 
+
+Refer to [Go Templates](https://pkg.go.dev/text/template) for more details.
+
+*Note:<br> if secretTemplate is used, the secretKey and secretRootKey attributes are ignored if provided.*
+
+
+Input:
+
+```yaml
+apiVersion: services.cloud.sap.com/v1
+kind: ServiceBinding
+metadata:
+  name: sample-binding
+spec:
+  serviceInstanceName: sample-instance
+  secretTemplate: your-Go-template
+```
+your-Go-template:
+
+```yaml
+apiVersion: services.cloud.sap.com/v1
+kind: ServiceBinding
+metadata:
+  name: sample-binding
+spec:
+  serviceInstanceName: sample-instance
+  secretTemplate: |
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      labels:
+        service_plan: {{ .instance.plan }}
+      annotations:
+        instance: {{ .instance.instance_name }}
+    stringData:
+      username: {{ .credentials.client_id }}
+      password: {{ .credentials.client_secret }}
+```
+The sample-binding map on which your-Go-template operates is in the JSON format:
+
```

Comment:
not sure I understand...

Reply:
@avilupu  I think we are still missing info here. Go template is applied to a data structure to generate an output. In this case, the generated output is the Secret format that we want. However, there is also a map. We need to explain more where this map comes from. @I065450